source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -426,7 +540,56 @@ TEST_F(NiFgenDriverApiTest, ResetInterchangeCheck_ResetsSuccessfully)
   ::grpc::Status status = GetStub()->ResetInterchangeCheck(&context, request, &response);
 
   EXPECT_TRUE(status.ok());
-  EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiFgenDriverApiTest, SendSoftwareEdgeTrigger_TriggersSuccessfully)
+{
+  ::grpc::ClientContext context;
+  fgen::SendSoftwareEdgeTriggerRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_trigger(fgen::Trigger::TRIGGER_NIFGEN_VAL_SCRIPT_TRIGGER);
+  request.set_trigger_id("ScriptTrigger0");
+  fgen::SendSoftwareEdgeTriggerResponse response;
+  ::grpc::Status status = GetStub()->SendSoftwareEdgeTrigger(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiFgenDriverApiTest, AllocateNamedWaveform_WriteNamedWaveformF64_WaveformWrittenSuccessfully)
+{
+  std::string channel_name = "0";
+  std::string waveform_name = "TestWaveform";
+  int waveform_size = 5;
+  allocate_named_waveform(channel_name, waveform_name, waveform_size);  
+  
+  double waveform[] = {1.55, 40.4, 21.6, 0.7, 15.89};
+  write_named_waveform_f64(channel_name, waveform_name, waveform_size, waveform);
+}
+
+TEST_F(NiFgenDriverApiTest, ExportTriggerMode_ResetAndImportConfiguration_TriggerModeConfigurationIsImported)
+{
+  const char* channel_name = "0";
+  fgen::TriggerMode expected_trigger_mode  = fgen::TriggerMode::TRIGGER_MODE_NIFGEN_VAL_SINGLE;
+  configure_trigger_mode(channel_name, expected_trigger_mode );
+  auto exported_configuration_response = export_attribute_configuration_buffer();
+
+  reset();
+  import_attribute_configuration_buffer(exported_configuration_response);
+
+  double set_trigger_mode = get_int32_attribute(channel_name, fgen::NiFgenAttributes::NIFGEN_ATTRIBUTE_TRIGGER_MODE);
+  EXPECT_EQ(expected_trigger_mode, set_trigger_mode);
+}
+
+TEST_F(NiFgenDriverApiTest, AllocateWaveform_WriteWaveformComplexF64_WaveformWrittenSuccessfully)
+{
+  std::string channel_name = "0";
+  int waveform_size = 1;
+  int waveform_handle = allocate_waveform(channel_name, waveform_size);  
+
+  NIComplexNumber waveform[] = {{1.55, 2.3}, {40.4, -20.4}, {21.6, 112.4}, {0.7, -100.3}, 15.89};
+  write_waveform_complexf64(channel_name, waveform_size, waveform_handle, waveform);
```

Comment:
Same as another comment above, return `response` object from `write_waveform_complexf64` helper, and do validation of `response.status()` here in separate section.